patternfly/patternfly-elements
# feat(core): add listbox support for roving-tabindex-controller

## What I did

1. Added directional support to `roving-tabindex-controller`.
2. Added a `listbox-controller`.
3. Added listbox components in order to test the support and demo how we can create an accessible listbox component: `pf-listbox`, `pf-listbox-group`, and `pf-listbox-option`.

## Testing Instructions

1. See [default demo](https://deploy-preview-2563--patternfly-elements.netlify.app/components/listbox/demo/)
2. See [orientation demo](https://deploy-preview-2563--patternfly-elements.netlify.app/components/listbox/demo/orientation/)
4. See [multiple selections demo](https://deploy-preview-2563--patternfly-elements.netlify.app/components/listbox/demo/multiple-selections/)
5. See [grouped options demo](https://deploy-preview-2563--patternfly-elements.netlify.app/components/listbox/demo/grouped-options/)
6. See [filter modes demo](https://deploy-preview-2563--patternfly-elements.netlify.app/components/listbox/demo/filter-modes/) (combobox support)

## Notes to Reviewers

**_This is in draft with a temporary listbox for testing._** We may be able to either use it as a reference or adapt it for a future listbox component.

Components with `role=listbox` can be used as standalone elements, as part of a combobox. See the following references:
1. [Toggle and Items Accessibility](https://docs.google.com/document/d/1lvAUBaBNiSiHDkWh3jXHDGOGnfmdOdIkWlxrBHrtgLc/edit)
7. [Web component benchmarking](https://docs.google.com/spreadsheets/d/1_mUVGwYBOuH1T5_vtc0us-CLnLUspizdEy5HXwVICvA/edit#gid=2066730968)
8. [WAI ARIA Listbox Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/)
9. [WAI ARIA Example Listbox Grouped](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/examples/listbox-grouped/)
10. [WAI ARIA Combobox Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/)
